edit/lint.js

```diff
@@ -0,0 +1,319 @@
+    event.preventDefault();
+    const json = tryJSONparse(popup.codebox.getValue());
+    if (json && json.rules) {
+      setStylelintRules(json.rules);
+      // it is possible to have stylelint rules popup open & switch to csslint
+      if (prefs.get('editor.linter') === 'stylelint') {
+        updateLinter('stylelint');
+      }
+    } else {
+      $('#help-popup .error').classList.add('show');
+      clearTimeout($('#help-popup .contents').timer);
+      $('#help-popup .contents').timer = setTimeout(() => {
+        // popup may be closed at this point
+        const error = $('#help-popup .error');
+        if (error) {
+          error.classList.remove('show');
+        }
+      }, 3000);
+    }
+  });
+  popup.querySelector('.reset').addEventListener('click', event => {
+    event.preventDefault();
+    setStylelintRules();
+    popup.codebox.setValue(JSON.stringify({rules: stylelintDefaultConfig.rules}, null, 2));
+    if (prefs.get('editor.linter') === 'stylelint') {
+      updateLinter('stylelint');
+    }
+  });
+}
+
+function openStylelintSettings() {
+  BG.chromeLocal.getValue('editorStylelintRules').then((rules = stylelintDefaultConfig.rules) => {
+    const rulesString = JSON.stringify({rules: rules}, null, 2);
+    setupStylelintPopup(rulesString);
+  });
+}
+
+function setupStylelintPopup(rules) {
+  function makeButton(className, text) {
+    return $element({tag: 'button', className, type: 'button', textContent: t(text)});
+  }
+  function makeLink(url, textContent) {
+    return $element({tag: 'a', target: '_blank', href: url, textContent});
+  }
+  function setJSONMode(cm) {
+    cm.setOption('mode', 'application/json');
+    cm.setOption('lint', 'json');
+  }
+  const popup = showCodeMirrorPopup(t('setStylelintRules'), $element({
+    appendChild: [
+      $element({
+        tag: 'p',
+        appendChild: [
+          t('setStylelintLink') + ' ',
+          makeLink('https://stylelint.io/demo/', 'Stylelint')
+        ]
+      }),
+      makeButton('save', 'styleSaveLabel'),
+      makeButton('reset', 'resetStylelintRules'),
+      $element({
+        tag: 'span',
+        className: 'error',
+        textContent: t('setStylelintError')
+      })
+    ]
+  }));
+  const contents = popup.querySelector('.contents');
+  const loadJSON = window.jsonlint ? [] : [
+    'vendor/codemirror/mode/javascript/javascript.js',
+    'vendor/codemirror/addon/lint/json-lint.js',
+    'vendor/jsonlint/jsonlint.js'
+  ];
+  contents.insertBefore(popup.codebox.display.wrapper, contents.firstElementChild);
+  popup.codebox.focus();
+  popup.codebox.setValue(rules);
+  onDOMscripted(loadJSON).then(() => { setJSONMode(popup.codebox); });
+  setupStylelintSettingsEvents(popup);
+}
+
+function loadSelectedLinter(name) {
+  let scripts = [];
+  if (name !== 'null' && !$('script[src*="css-lint.js"]')) {
+    // inject css
+    injectCSS('vendor/codemirror/addon/lint/lint.css');
+    // load CodeMirror lint code
+    scripts = scripts.concat([
```

Comment:
LOL, I never realized that you can `push` more than one item <img title=":_ derp :" src="https://cloud.githubusercontent.com/assets/136959/13559387/49e2ecce-e3d9-11e5-8a16-73cbfdb7230e.gif">